### ISSUE TITLE
[Bug]: fallback to settings prompt is shown when calling requestPermission(false)

### ISSUE THREAD

**Issue by Redn4s** [OneSignal/react-native-onesignal#1613]
### What happened?

When requesting the push permission after the user already declined push permissions, the "Open settings" prompt is still shown even when I pass `false` when calling `requestPermission()`.

`OneSignal.Notifications.requestPermission(false) // Still shows the "Open settings" alert`

### Steps to reproduce?

```Markdown
1. Install v5.0.4 of react-native-onesignal
2. Call `requestPermission(false)`
3. Decline the push permission
4. Call `requestPermission(false)` again
5. The "Open settings" prompt is shown
```

### What did you expect to happen?

I expect the "Open settings" prompt not to show up when I call `requestPermission(false)` after the user already declined push permissions.

### React Native OneSignal SDK version

5.0.4

### Which platform(s) are affected?

- [X] iOS
- [ ] Android

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ded3zito** [comment]
I was also getting this error and the problem was that I was calling `requestPermission()` twice inside the codebase. calling it only once solved the problem.

**Comment by BijuDey** [comment]
> I was also getting this error and the problem was that I was calling `requestPermission()` twice inside the codebase. calling it only once solved the problem.

I have called it just once still same issue

**Comment by unknwngera** [comment]
```
OneSignal.Notifications.canRequestPermission().then(can => {
  if (can) OneSignal.Notifications.requestPermission(false)
})
```

solves this problem

**Comment by Anurodhydv** [comment]
having the same issue. fall back prompt appears on route change as well.

**Comment by fadi-george** [comment]
Hello,
I tried to replicate this issue on the latest version but was not able to reproduce it. I tested both on Android and iOS. If anyone has a similar issue on the latest version, please open a new issue with more details.
